script.lua

```diff
@@ -3230,57 +3230,97 @@ function setupPlayerArea(params)
 
     local energy = 0
 
-    local function elemStrToArr(elemStr)
-        local outArr = {}
-        for i = 1, string.len(elemStr) do
-            table.insert(outArr,(math.floor(string.sub(elemStr, i, i))))
-        end
-        return outArr
-    end
-
-    local function elemCombine(inTableOfElemStrCards)
+    local Elements = {}
+    Elements.__index = Elements
+    function Elements:new(init)
         local outTable = {0,0,0,0,0,0,0,0}
-        for i = 1, #inTableOfElemStrCards do
-            local elemTable = elemStrToArr(inTableOfElemStrCards[i].getVar("elements"))
+        setmetatable(outTable, self)
+        outTable:add(init)
+        return outTable
+    end
+    function Elements:add(other)
+        if other == nil then
+            return
+        elseif type(other) == "table" then
             for j = 1, 8 do
-                outTable[j] = outTable[j] + elemTable[j]
+                self[j] = self[j] + other[j]
+            end
+        elseif type(other) == "string" then
+            for i = 1, string.len(other) do
+                self[i] = self[i] + math.floor(string.sub(other, i, i))
             end
-            local cost = inTableOfElemStrCards[i].getVar("energy")
-            -- Skip counting locked card's energy (Aid from Lesser Spirits)
-            if not inTableOfElemStrCards[i].getLock() and cost ~= nil then
-                energy = energy + cost
-                if (inTableOfElemStrCards[i].hasTag("Fast") and not inTableOfElemStrCards[i].hasTag("Temporary Slow")) or inTableOfElemStrCards[i].hasTag("Temporary Fast") then
-                    energy = energy - fastDiscount
+        end
+    end
+    function Elements:__tostring()
+        return table.concat(self, "")
+    end
+
+    local function powerCost(card)
+        local cost = card.getVar("energy")
+        -- Skip counting locked card's energy (Aid from Lesser Spirits)
+        if card.getLock() or cost == nil then
+            return 0
+        end
+        if (card.hasTag("Fast") and not card.hasTag("Temporary Slow")) or card.hasTag("Temporary Fast") then
+            cost = cost - fastDiscount
+        end
+        return cost
+    end
+
+    local function calculateTrackElements(spiritBoard)
+        local elements = Elements:new()
+        if spiritBoard.script_state ~= "" then
+            local trackElements = spiritBoard.getVar("trackElements")
+            if trackElements ~= nil then
+                for _, trackElem in pairs(trackElements) do
+                    local hits = Physics.cast{
+                        origin = spiritBoard.positionToWorld(trackElem.position), -- pos
+                        direction = Vector(0, 1, 0),
+                        max_distance = 1,
```

Comment:
max_distance of 1 seems a bit high, can this value be lower?

Reply:
I don't think it is worth optimizing this.